File: 0724_find_pivot_index/prefix_sum/clojure/src/solution.clj
(ns solution)

(defn pivot-index [nums]
  (let [total-sum (reduce + nums)]
    (loop [left-sum 0
           idx 0
           [num & remaining] nums]
      (if (nil? num)
        -1
        (if (= left-sum (- total-sum left-sum num))
          idx
          (recur (+ left-sum num) (inc idx) remaining))))))

File: 0724_find_pivot_index/prefix_sum/go/solution.go
package pivot_index

func pivotIndex(nums []int) int {
    totalSum := 0
    for _, num := range nums {
        totalSum += num
    }
    
    leftSum := 0
    for i, num := range nums {
        if leftSum == totalSum - leftSum - num {
            return i
        }
        leftSum += num
    }
    
    return -1
}
